### DISCUSSION TITLE
Implementing glowroot in an elasticsearch image 7.17

### DISCUSSION THREAD

**Discussion by azoot21** [glowroot/glowroot#1101] [General]
Hello,

I am currently deploying the Glowroot agent in a Docker image of Elasticsearch 7.17. Elasticsearch comes with its own version of Java 17.

I encountered several issues related to access rights to my JVM information, but I managed to resolve them by granting various MBean access permissions in a Java security file.

However, now I'm facing one last error:

`ERROR o.g.a.w.WeavingClassFileTransformer - error weaving sun/security/util/math/intpoly/IntegerPolynomial$MutableElement: sun/security/util/math/intpoly/IntegerPolynomial$MutableElement`

I tried installing different versions of Glowroot, as I thought it might be a compatibility issue with the Java version used by my Elasticsearch, but to no avail.

Does anyone have any ideas, please?

Thanks in advance.

**Comment by Never-lab** [comment]
That `WeavingClassFileTransformer` error on `sun/security/util/math/intpoly/...` is the agent attempting to weave a **JDK internal** class inside Elasticsearch’s bundled JVM. In custom/tight JVM images this often logs as ERROR but is **usually non-fatal** — the class is left unweaved and the agent continues.

Worth checking:
1. **Latest stable agent** (0.14.6) — several JDK 17+ edge cases were fixed since older builds.
2. Does the Glowroot UI still come up despite the log line? If yes, you may be able to proceed.
3. Confirm you really need `-javaagent` on the **Elasticsearch JVM** itself. Most setups monitor the **application** JVM; ES-specific instrumentation is via the elasticsearch plugin on the app side.

If the agent fails to start entirely, please add Glowroot version, full `-javaagent` args, and startup logs — reproducible failures belong in **Issues**. For troubleshooting like this, **Q&A** is the better category (#1223).
